src/Features/Core/Portable/ExtractMethod/MethodExtractor.cs

```diff
@@ -89,6 +97,66 @@ public async Task<ExtractMethodResult> ExtractMethodAsync(CancellationToken canc
                 generatedCode.MethodNameAnnotation,
                 generatedCode.MethodDefinitionAnnotation,
                 cancellationToken).ConfigureAwait(false);
+
+            bool CanAddTo(Document document, SyntaxNode insertionPointNode, out OperationStatus status)
+            {
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var syntaxKinds = syntaxFacts.SyntaxKinds;
+                var codeGenService = document.GetLanguageService<ICodeGenerationService>();
+
+                if (insertionPointNode is null)
+                {
+                    status = OperationStatus.NoValidLocationToInsertMethodCall;
+                    return false;
+                }
+
+                var destination = insertionPointNode;
+                if (!LocalFunction)
+                {
+                    var mappedPoint = insertionPointNode.RawKind == syntaxKinds.GlobalStatement
+                        ? insertionPointNode.Parent
+                        : insertionPointNode;
+                    destination = mappedPoint.Parent ?? mappedPoint;
```

Comment:
this is acopy of that strange logic seen above.  i will try in a followup to push this into GetInsertionPointNode and make it so that we don't need this massaging here and in the code generator.